utils/atomic/funcs.go

```diff
@@ -0,0 +1,13 @@
+package atomic
+
+import (
+	"sync/atomic"
+)
+
+/*
+TODO: move to sdk-utils
+*/
+
+func Uint64Sub(x *atomic.Uint64, y uint64) uint64 {
```

Comment:
can you add a comment mentioning that this uses two's complement arithmetic

Reply:
implementation detail, why is that relevant

Reply:
because not everyone might be familiar with how your function works

Reply:
also, once again, can you please leave the conversation resolving to the reviewer

Reply:
TBH I didn't know what twos complement is so the comment wouldn't have helped me
I just did an example on paper 
Not sure what benefit you get from 'twos complement' because if you understand twos complement you don't need the comment
And if you don't understand twos complement you will have to work out the math anyway

Reply:
I mean, some docstring would be nice

Reply:
done

Reply:
the expression ^(y - 1) flips all the bits of the result of (y - 1), which gives you -y
then you just need to do addition with that negative number. I assume you did it like this since atomic package doesn't have a Sub method

Reply:
>the expression ^(y - 1) flips all the bits of the result of (y - 1), which gives you -y
then you just need to do addition with that negative number.

I understand that, this is just repeating the code in english

> I assume you did it like this since atomic package doesn't have a Sub method

👍